### PR TITLE
[headers 5] Misc small headers untangling

### PR DESCRIPTION
**Pull request by Dragorn421**
Working towars untangling headers for the purpose of z_fishing bss
This all follows what's already in MM

### CODE REVIEW

**Comment by Dragorn421** [comment]
`src/audio/general.c` bss doesn't match in gc-eu-mq-dbg: the block number for bss symbol `sAudioUpdateTaskEnd` wraps to 0 (mod 256) when it's at the end of bss in the baserom.

I have no idea how to work around this in debug, where our #pragma increment_block_number mechanism doesn't work.
I guess we need this file to include less headers than global.h: this needs at least #2151 for `gPadMgr` to be moved out of variables.h

drafting until #2151 is merged

File: include/sequence.h [hunk 1]
```diff
@@ -323,4 +325,18 @@ typedef enum NatureAmimalId {
 
 #define NATURE_IO_ENTRIES_END 0xFF
 
+// functions in sequence.c
+
```
Comment (cadmic) [inline include/sequence.h:329]: Do you plan to add more function declarations to this file? sequence.c seems to be the default place to look anyway, so I don't think this comment is necessary

File: include/z64save.h [hunk 2]
```diff
@@ -892,4 +892,6 @@ typedef enum EventInfHorsesState {
```
Comment (cadmic) [inline include/z64save.h:894]: 3 blank lines feels a bit excessive here

File: include/sequence.h (hunk 1 above)
Comment (Dragorn421) [reply include/sequence.h:329]: Yes MM has more

File: src/overlays/actors/ovl_En_Kanban/z_en_kanban.c [hunk 3]
```diff
@@ -5,6 +5,7 @@
  */
 
 #include "z_en_kanban.h"
+#include "global.h"
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Kanban/z_en_kanban.c:8]: Why did you add global.h here and in other files?

Comment (Dragorn421) [reply src/overlays/actors/ovl_En_Kanban/z_en_kanban.c:8]: in main, z_fishing indirectly includes global.h through including  z_en_kanban.h

As a first step towards not having to deal with z_fishing bss every single PR ever, I explicitly include global.h in z_fishing.h (goal: remove it eventually) and remove it from z_en_kanban.h (but have to add it back to z_en_kanban.c)

**Review by AngheloAlf — APPROVED** [review]

**Review by cadmic — APPROVED** [review]

File: include/z64math.h [hunk 4]
```diff
@@ -145,4 +145,7 @@ typedef VecSphGeo VecGeo;
 #define SQXYZ(vec) ((vec).x * (vec).x + (vec).y * (vec).y + (vec).z * (vec).z)
 #define DOTXYZ(vec1, vec2) ((vec1).x * (vec2).x + (vec1).y * (vec2).y + (vec1).z * (vec2).z)
 
+f32 Rand_ZeroFloat(f32 f);
+f32 Rand_CenteredFloat(f32 f);
```
Comment (fig02) [inline include/z64math.h:149]: initial reaction is that this doesnt fit here in math.h, but I understand the dillema here. The functions live in z_actor and are used by more then actors. We probably cant put it with the other rand functions cause those are from an external library right

Comment (Dragorn421) [reply include/z64math.h:149]: yeah I have the same train of thoughts.
(note this location is copied from MM)

Maybe we could create a new rand.h and rename existing rand.h to librand.h ?

Comment (fig02) [reply include/z64math.h:154]: possibly
I also remember some people wanting to use the original function names for the library functions.
A game-side header could give the opportunity for both options to coexist: The library uses the real names, and we give them a wrapper define on the game-side with a name we like

Comment (Dragorn421) [reply include/z64math.h:154]: #2164

**Review by cadmic — APPROVED** [review]
seems good to me (unsure if this still "Needs discussion")

**Review by fig02 — APPROVED** [review]
